### PR TITLE
Animate selection change

### PR DESCRIPTION
**Pull request by andris-zalitis**

### CODE REVIEW

**Comment by andris-zalitis** [comment]
The animation can be seen if checking out branch for this PR: https://github.com/worldslastchance/wlc-calendar/pull/73

File: SlidingSegmentedControlTests/SlidingSegmentedControlTests.swift [hunk 1]
```diff
@@ -507,3 +639,37 @@ struct SegmentLocatorStub: SegmentLocator {
...
+
+    static func animate(withDuration duration: TimeInterval, delay: TimeInterval, usingSpringWithDamping dampingRatio: CGFloat, initialSpringVelocity velocity: CGFloat, options: UIViewAnimationOptions, animations: @escaping () -> Void, completion: ((Bool) -> Swift.Void)?) {
+        let params = EquatableParams(duration: duration, delay: delay, dampingRatio: dampingRatio, velocity: velocity, options: options)
+        capturedEquatableParamsArray.append(params)
+        capturedAnimationsArray.append(animations)
+    }
+
+}
+
+func == (lhs: AnimatingMock.EquatableParams, rhs: AnimatingMock.EquatableParams) -> Bool {
```
Comment (rudolf-adamkovic) [inline SlidingSegmentedControlTests/SlidingSegmentedControlTests.swift]: Wouldn't `String(describing: lhs) == String(describing: rhs)` be enough?

File: SlidingSegmentedControlTests/SlidingSegmentedControlTests.swift [hunk 2]
```diff
@@ -507,3 +639,37 @@ struct SegmentLocatorStub: SegmentLocator {
...
+    struct EquatableParams: Equatable {
+        let duration: TimeInterval
+        let delay: TimeInterval
+        let dampingRatio: CGFloat
+        let velocity: CGFloat
+        let options: UIViewAnimationOptions
+    }
+
+    static var capturedEquatableParamsArray: [EquatableParams] = []
+    static var capturedAnimationsArray: [() -> Void] = []
```
Comment (rudolf-adamkovic) [inline SlidingSegmentedControlTests/SlidingSegmentedControlTests.swift]: Consider removing unnecessary type repetition (`Array`)

File: SlidingSegmentedControlTests/SlidingSegmentedControlTests.swift [hunk 3]
```diff
@@ -507,3 +639,37 @@ struct SegmentLocatorStub: SegmentLocator {
...
+
+    struct EquatableParams: Equatable {
+        let duration: TimeInterval
+        let delay: TimeInterval
+        let dampingRatio: CGFloat
+        let velocity: CGFloat
+        let options: UIViewAnimationOptions
+    }
+
+    static var capturedEquatableParamsArray: [EquatableParams] = []
```
Comment (rudolf-adamkovic) [inline SlidingSegmentedControlTests/SlidingSegmentedControlTests.swift]: Consider removing unnecessary type repetition (`Array`) and improving non-descriptive name (`equatableParams`).

**Review by rudolf-adamkovic — CHANGES_REQUESTED** [review]

**Comment by andris-zalitis** [comment]
@salutis great points as usual! I've addressed these, ready for review again.

**Comment by rudolf-adamkovic** [comment]
@andris-zalitis What about `capturedConfiguration` or `capturedParameters` instead of `capturedConfigurationParams`?

**Comment by andris-zalitis** [comment]
@salutis I was just thinking about it this morning - "why did I name them params?" 😄 
BTW you can't name them `capturedParameters` because they are only for configuration and not for animation and completion parameters.

**Comment by andris-zalitis** [comment]
@salutis 
Ready for review

**Review by rudolf-adamkovic — APPROVED** [review]

**Comment by rudolf-adamkovic** [comment]
@andris-zalitis Top-notch work!

**Comment by andris-zalitis** [comment]
@salutis Thank you for review and merge!

**Pull request merged**